leakcanary-deobfuscation-gradle-plugin/src/test/java/com/squareup/leakcanary/deobfuscation/CopyObfuscationMappingFileTaskTest.kt

```diff
@@ -0,0 +1,93 @@
+package com.squareup.leakcanary.deobfuscation
+
+import org.assertj.core.api.Assertions.assertThat
+import org.assertj.core.api.Assertions.assertThatExceptionOfType
+import org.gradle.api.GradleException
+import org.gradle.testfixtures.ProjectBuilder
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.TemporaryFolder
+import java.io.File
+
+class CopyObfuscationMappingFileTaskTest {
```

Comment:
This test tests well the task. Do we want to add one to test the logic of the plugin ? It would help maintainers, especially as they are less familiar to gradle plugins.

Reply:
You mean something like this? https://github.com/gradle-guides/testing-gradle-plugins/blob/master/samples/code/url-verifier-plugin/src/functTest/groovy/org/gradle/sample/UrlVerifierPluginFunctionalTest.groovy

I've thought about adding such test but we rely on Android plugin. I'm not sure how would that work, do we need to have a sample android project that will be used during the test to compile and generate mapping file? Can we mock it somehow? I couldn't find anything useful on the web.

Reply:
yes, that's the idea. for an android project, you also need a manifest, and one source file, nothing else IIRC.

Reply:
I've added the test, but I'm not sure how to properly pass android sdk path to the test so it's able to perform a build on an android project.